### PR TITLE
refactor/1684 parser directory layout

### LINKED ISSUE #1684: [refactor] improve command parser
I took a look at the command parser and noticed that with some work, we could save a lot of LOC while improving readability :)
To make the review process easier, I'm going to submit the PRs individually.

The first step is to refactor the directory layout

### PR DESCRIPTION
**Pull request by JonasCir**
REPLACED BY #1691.

Closes #1684 

This pull request does:

- move the `parsers` folder into a new `commands` folder. Commands drive the simulation, not parsers :)
- renamed `AicraftCommandParser` to `CommandParser` as it parses system commands like `tutorial` as well

By this we can start to separate commands and the parser.

---

Test, build, and gameplay passes.

### CODE REVIEW

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 import { EVENT } from '../constants/eventNames';
 import { radians_normalize } from '../math/circle';
 import { round } from '../math/core';
-import { AIRCRAFT_COMMAND_MAP } from '../parsers/aircraftCommandParser/aircraftCommandMap';
+import { AIRCRAFT_COMMAND_MAP } from '../commands/aircraftCommand/AircraftCommandMap';
```
Comment (erikquinn) [inline src/assets/scripts/client/aircraft/AircraftCommander.js:17]: given that it's not a class, `aircraftCommandMap.js` should not be capitalized.

File: src/assets/scripts/client/commands/parsers/ArgumentParsers.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import _defaultTo from 'lodash/defaultTo';
-import { isValidCourseString, isValidDirectionString } from './argumentValidators';
+import { isValidCourseString, isValidDirectionString } from './ArgumentValidators';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/parsers/ArgumentParsers.js:2]: lower case, due to not being a class

File: src/assets/scripts/client/commands/parsers/ArgumentValidators.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import _isNaN from 'lodash/isNaN';
 import _isString from 'lodash/isString';
 import _forEach from 'lodash/forEach';
 import { convertStringToNumber } from '../../utilities/unitConverters';
-import { EXPEDITE } from './aircraftCommandMap';
-import { ERROR_MESSAGE } from './aircraftCommandParserMessages';
+import { EXPEDITE } from '../aircraftCommand/AircraftCommandMap';
+import { ERROR_MESSAGE } from './ParserMessages';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/parsers/ArgumentValidators.js:6]: lower case, due to not being a class

File: src/assets/scripts/client/commands/aircraftCommand/AircraftCommandDefinitions.js [hunk 4]
```diff
@@ -21,15 +21,15 @@ import {
     squawkValidator,
     optionalAltitudeValidator,
     crossingValidator
-} from './argumentValidators';
+} from '../parsers/ArgumentValidators';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/aircraftCommand/AircraftCommandDefinitions.js:24]: lower case, due to not being a class

File: src/assets/scripts/client/commands/aircraftCommand/AircraftCommandDefinitions.js [hunk 5]
```diff
@@ -21,15 +21,15 @@ import {
...
 import {
     altitudeParser,
     headingParser,
     holdParser,
     timewarpParser,
     optionalAltitudeParser,
     crossingParser
-} from './argumentParsers';
+} from '../parsers/ArgumentParsers';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/aircraftCommand/AircraftCommandDefinitions.js:32]: lower case, due to not being a class

File: src/assets/scripts/client/commands/aircraftCommand/AircraftCommandModel.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { AIRCRAFT_COMMAND_DEFINITION } from './aircraftCommandDefinitions';
+import { AIRCRAFT_COMMAND_DEFINITION } from './AircraftCommandDefinitions';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/aircraftCommand/AircraftCommandModel.js:1]: lower case, due to not being a class

File: src/assets/scripts/client/commands/parsers/ArgumentValidators.js [hunk 7]
```diff
@@ -2,8 +2,8 @@ import _isNaN from 'lodash/isNaN';
 import _isString from 'lodash/isString';
 import _forEach from 'lodash/forEach';
 import { convertStringToNumber } from '../../utilities/unitConverters';
-import { EXPEDITE } from './aircraftCommandMap';
-import { ERROR_MESSAGE } from './aircraftCommandParserMessages';
+import { EXPEDITE } from '../aircraftCommand/AircraftCommandMap';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/parsers/ArgumentValidators.js:5]: lower case, due to not being a class

File: src/assets/scripts/client/commands/parsers/CommandParser.js [hunk 8]
```diff
@@ -3,11 +3,11 @@ import _forEach from 'lodash/forEach';
 import _isString from 'lodash/isString';
 import _map from 'lodash/map';
 import _tail from 'lodash/tail';
-import AircraftCommandModel from './AircraftCommandModel';
+import AircraftCommandModel from '../aircraftCommand/AircraftCommandModel';
 import {
     AIRCRAFT_COMMAND_MAP,
     findCommandNameWithAlias
-} from './aircraftCommandMap';
+} from '../aircraftCommand/AircraftCommandMap';
```
Comment (erikquinn) [inline src/assets/scripts/client/commands/parsers/CommandParser.js:10]: lower case, due to not being a class

**Comment by JonasCir** [comment]
@erikquinn Tank you for the fast review and sorry for that, I am new to JS and still learning :) Should be good to go now

**Review by erikquinn — APPROVED** [review]

**Review by n8rzz — APPROVED** [review]
Seems I was logged in as @3tractorrides a bit ago.  Whoops, thats my work account.  Reach out to me on this one, @n8rzz.

:shipit:

**Comment by JonasCir** [comment]
@n8rzz alright will do :smiley:

**Comment by erikquinn** [comment]
replaced by #1691. closing.

**Pull request closed without merging**